httpmw/rbac.go

```diff
@@ -0,0 +1,27 @@
+package httpmw
+
+import (
+	"log"
+
+	"github.com/gin-gonic/gin"
+	"github.com/protsack-stephan/gin-toolkit/httperr"
+)
+
+// RBACAuthorizeFunc is the type alias for a RBAC Authorize function signature.
+type RBACAuthorizeFunc func(*gin.Context) (bool, error)
+
+// RBAC implements RBAC using the provided authorizer function.
+func RBAC(fn RBACAuthorizeFunc) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if authorized, err := fn(c); err != nil || !authorized {
+			if err != nil {
+				log.Println(err)
+			}
+			httperr.Unauthorized(c)
+			c.Abort()
+			return
+		}
```

Comment:
I suggest changing the logic a little bit, let's not print the error but throw internal server error instead, something like:
```go
ok, err := fn(c)

if err != nil {
	httperr.InternalServerError(c, err.Error())
	c.Abort()
	return
}

if !ok {
	httperr.Unauthorized(c)
	c.Abort()
	return
}

c.Next()
```

Reply:
Makes sense. Done